application/helpers/frontend_helper.php

```diff
@@ -1886,13 +1889,39 @@ function doAssessment($surveyid, $returndataonly=false)
                             if ($usresult)
                             {
                                 $usrow = $usresult->read();
-                                $fieldmap[$field['fieldname']]['assessment_value']=$usrow['assessment_value'];
-                                $total=$total+$usrow['assessment_value'];
+                                $assessmentValue = $usrow['assessment_value'];
                             }
                         }
                     }
                     $groups[]=$field['gid'];
                 }
+
+                // If this is a question (and not a survey field, like ID), save asessment value
+                if (!empty($field['title']))
+                {
+                    /**
+                     * Allow Plugin to update assessment value
+                     */
+                    // Prepare Event Info
+                    $event = new PluginEvent('afterSurveyQuestionAssessment');
+                    $event->set('surveyId', $surveyid);
+                    $event->set('question', $field);
```

Comment:
Thinking, for question/subquestion (and answer)

- Allways send `surveyid` as `surveyid` 
- Seems best to send `qid` as `qid` for PARENT qid (the one you can edit directly in admin gui)
- For subquestion : didn't know for sure : title or qid + scale_id ? 
- For 'database column'/'answer' : `sgq` seems good (else we need to [construct](https://framagit.org/SondagePro-LimeSurvey-plugin/arrayTextAdapt/blob/master/arrayTextAdapt.php#L156) each time)
- Seems interseting to have question type (but not mandatory)

It's the first plugin event for SGQ, then we must choose seriously. I really want a plugin event in [ExpressionManager _validateQuestion](https://github.com/LimeSurvey/LimeSurvey/blob/master/application/helpers/expressions/em_manager_helper.php#L6095) : then this plugin fix some part we send in next pluginb for SGQA